src/Shape/Whisker.js

```diff
@@ -0,0 +1,257 @@
+import {select} from "d3-selection";
+
+import {accessor, assign, BaseClass, configPrep, constant, elem} from "d3plus-common";
+
+import Circle from "./Circle";
+import Line from "./Line";
+import Rect from "./Rect";
+
+/**
+    @class Whisker
+    @extends BaseClass
+    @desc Creates SVG lines based on an array of data.
+*/
+export default class Whisker extends BaseClass {
+
+  /**
+      @memberof Whisker
+      @desc Invoked when creating a new class instance, and overrides any default parameters inherited from BaseClass.
+      @private
+  */
+  constructor() {
+
+    super();
+
+    this._endpoint = constant("Circle");
+    this._endpointConfig = {
+      Circle: {
+        fill: d => d.fill,
+        r: d => d.r
+      },
+      Rect: {
+        fill: d => d.fill,
+        height: d => d.height,
+        stroke: constant("black"),
+        strokeWidth: constant(1),
+        width: d => d.width
+      }
+    };
+    this._length = accessor("length", 25);
+    this._lineConfig = {
+      stroke: constant("black"),
+      strokeWidth: constant(1)
+    };
+    this._name = "Whisker";
+    this._orient = accessor("orient", "top");
+    this._shapes = [];
+    this._x = accessor("x", 0);
+    this._y = accessor("y", 0);
+
+  }
+
+  /**
+      @memberof Box
+      @desc Filters/manipulates the data array before binding each point to an SVG group.
+      @param {Array} [*data* = the data array to be filtered]
+      @private
+  */
+  _whiskerData(data) {
```

Comment:
This method of splitting the data into separate shapes is not sustainable, as it will require to manual add each new shape as it's created (for example, if we wanted to support a new `Triangle` shape).

Please look at how `Plot.js` separates it's data into different shapes. It's possible to have an x/y plot with any shape supported by `d3plus-shape`.
* The data is nested like this: https://github.com/d3plus/d3plus-plot/blob/master/src/Plot.js#L371
* The `key` of the nest is used to create each class: https://github.com/d3plus/d3plus-plot/blob/master/src/Plot.js#L684

If you are having trouble understanding how `d3.nest` works, please read this write-up and then come to me with any additional questions: http://learnjsdata.com/group_data.html